### PR TITLE
Y026: explicitly declare type aliases as TypeAlias

### LINKED ISSUE #345: Convert typing declarations to use TypeAlias
See discussion here:  https://github.com/pandas-dev/pandas-stubs/pull/341#issuecomment-1262928643

Should we change `_typing.pyi` to make all these types that we create be declared with `TypeAlias` (imported from `typing_extensions`) ?

@twoertwein - your opinion?

### PR DESCRIPTION
**Pull request by twoertwein**
Closes #345

(and remove `FilePathOrBuffer` - it does not exist anymore in pandas)

### CODE REVIEW

File: pandas-stubs/_version.pyi [hunk 1]
```diff
@@ -4,4 +4,4 @@ version_json: str = ...
 
 def get_versions(): ...
 
-_stub_version = Literal["1.5.0.220926"]
+_stub_version: Literal["1.5.0.220926"]
```
Comment (twoertwein) [inline pandas-stubs/_version.pyi:7]: Not sure why this is needed.

If we assign Literal as a value, we need a TypeAlias, but the type alias is unused  (private type aliases need to be used) which triggers another check from flake8-pyi.

**Comment by twoertwein** [comment]
Would be nice to slowly move all the TypeAliases that are not in _typing to something like __typing (no need to make things worse for #128)

File: pandas-stubs/_version.pyi (hunk 1 above)
Comment (Dr-Irv) [reply pandas-stubs/_version.pyi:7]: Microsoft is copying the stubs over to the VS Code pylance distribution, so by having this, we can tell what version they copied over.

**Review by Dr-Irv — APPROVED** [review]
thanks @twoertwein

**Pull request merged**